prov/util/src/util_av.c

```diff
@@ -333,7 +333,7 @@ static int fi_av_remove_addr(struct util_av *av, int slot, int index)
 	struct dlist_entry *av_entry;
 	int *entry, *next, i;
 
-	if (index < 0 || index > av->count) {
+	if (index < 0 || (size_t)index > av->count) {
```

Comment:
For all the changes in the file, would it make more sense to convert index to size_t instead of an int?